tools/script_manager.lua

```diff
@@ -0,0 +1,764 @@
+
+  result = os.execute(git_command)
+
+  if not df.check_if_file_exists(LUA_DIR .. PS .. "downloads") then
+    os.execute("mkdir " .. LUA_DIR .. PS .. "downloads")
+  end
+
+  return result
+end
+
+local function update_scripts()
+  local result = false
+
+  local git = df.check_if_bin_exists("git")
+
+  if not git then
+    dt.print("ERROR: git not found.  Install or specify the location of the git executable.")
+    return
+  end
+
+  local git_command = "cd " .. LUA_DIR .. " " .. CS .. " " .. git .. " pull"
+  dt.print_log("update git command is " .. git_command)
+
+  if dt.configuration.running_os == "windows" then
+    result = dtsys.windows_command(git_command)
+  else
+    result = os.execute(git_command)
+  end
+
+  if dt.preferences.read("script_manager", "use_lua_scripts_version", "bool") and 
+     dt.configuration.running_os == "windows" then
+    use_lua_scripts_version()
+  end
+
+  return result
+end
+
+local function add_script_data(script_file)
+
+  local pattern = "(.-)/(.-)(([^\\/]-)%.?([^%.\\/]*))$"
+  if dt.configuration.running_os == "windows" then
+    pattern = "(.-)\\(.-)(([^\\]-)%.?([^%.\\]*))$"
+  end
+
+  dt.print_log("processing " .. script_file)
+
+  -- add the script data
+  local category,path,name,filename,filetype = string.match(script_file, pattern)
+
+  if #sm.script_categories == 0 or not string.match(du.join(sm.script_categories, " "), category) then
+    sm.script_categories[#sm.script_categories + 1] = category
+    sm.script_names[category] = {}
+  end
+  if name then
+    if not string.match(du.join(sm.script_names[category], " "), name) then
+      sm.script_names[category][#sm.script_names[category] + 1] = name
+      sm.script_paths[category .. PS .. name] = category .. PS .. path .. name
+      if category == "downloads" then
+        sm.have_downloads = true
+      end
+    end
+  end
+end
+
+local function scan_scripts()
+  local find_cmd = "find -L " .. LUA_DIR .. " -name \\*.lua -print | sort"
+  if dt.configuration.running_os == "windows" then
+    find_cmd = "dir /b/s " .. LUA_DIR .. "\\*.lua | sort"
+  end
+  -- scan the scripts
+  local output = io.popen(find_cmd)
+  for line in output:lines() do
+    local l = string.gsub(line, LUA_DIR .. PS, "") -- strip the lua dir off
+    local script_file = l:sub(1,-5)
+    if not string.match(script_file, "script_manager") then  -- let's not include ourself
+      if not string.match(script_file, "plugins") then -- skip plugins
+        if not string.match(script_file, "lib" .. PS) then -- let's not try and run libraries
+          if not string.match(script_file, "include_all") then -- skip include_all.lua
+            if not string.match(script_file, "yield") then -- special case, because everything needs this
+              add_script_data(script_file)
+            else
+              prequire(script_file) -- load yield.lua
+            end
+          end
+        end
+      end
+    end
+  end
```

Comment:
Please consider using "and" instead of the nested "if". I think that is easier to read and the code would look cleaner.

Reply:
I like the and better too